R/readUNFCCC_NDC.R

```diff
@@ -64,18 +69,134 @@ readUNFCCC_NDC <- function(subtype, subset) {
+          `Conditional Absolute` =dplyr::if_else(
+            `Type of NDC` == "Specific" & Conditionality == "Conditional",
+            as.character(`emission level max (Mt CO2eq)...17`),
+            `Conditional Absolute`
+          ),
+          
+          # BAU unconditional
+          `Unconditional Relative` =dplyr::case_when(
+            `Type of NDC` == "BAU" & Conditionality == "Unconditional" & !is.na(`reduction min (%)...18`) ~ 
+              as.character(`reduction min (%)...18`),
+            TRUE ~ `Unconditional Relative`
+          ),
+          `Unconditional Absolute` =dplyr::case_when(
+            `Type of NDC` == "BAU" & Conditionality == "Unconditional" & is.na(`reduction min (%)...18`) ~ 
+              as.character(`emission level min (Mt CO2eq)...16`),
+            TRUE ~ `Unconditional Absolute`
+          ),
+          
+          # BAU conditional
+          `Conditional Relative` =dplyr::case_when(
+            `Type of NDC` == "BAU" & Conditionality == "Conditional" & !is.na(`reduction min (%)...18`) ~ 
+              as.character(`reduction min (%)...18`),
+            TRUE ~ `Conditional Relative`
+          ),
+          `Conditional Absolute` =dplyr::case_when(
+            `Type of NDC` == "BAU" & Conditionality == "Conditional" & is.na(`reduction min (%)...18`) ~ 
+              as.character(`emission level min (Mt CO2eq)...16`),
+            TRUE ~ `Conditional Absolute`
+          ),
+          
+          # correct the GHG type 
+          Type =dplyr::case_when(
+            `Gas coverage` == "GHG" & 
+              (!is.na(`Unconditional Relative`) | !is.na(`Conditional Relative`)) ~ "GHG",
+            #attention! In the current PBL file absolute emission targets always
+            #refer to emission level targeted in the target year, no reduction levels
+            `Gas coverage` == "GHG" & `Type of NDC` != "Specific" &
+              (!is.na(`Unconditional Absolute`) | !is.na(`Conditional Absolute`)) ~ "GHG-fixed-total",
+            
+            `Gas coverage` == "GHG" & `Type of NDC` == "Specific" &
+              (!is.na(`Unconditional Absolute`) | !is.na(`Conditional Absolute`)) ~ "GHG-fixed-total",
+            
+            TRUE ~ "type missing"
+          )
+        ) %>%
+        select(
+          ISO_Code,
+          Reference_Year,
+          BAU_or_Reference_emissions_in_MtCO2e,
+          Target_Year,
+          Type,
+          LULUCF,
+          `Unconditional Absolute`,
+          `Conditional Absolute`,
+          `Unconditional Relative`,
+          `Conditional Relative`
         )
+      
+      #set BAU where needed
+      input2035$Reference_Year[!is.na(input2035$BAU_or_Reference_emissions_in_MtCO2e) & 
+                                 input2035$Type != "GHG-fixed-total"] <- "BAU"
+      
+      #set as negative % targets (reduction)
+      input2035$`Unconditional Relative` <- as.numeric(input2035$`Unconditional Relative`) / -100
```

Comment:
minor comment: why not do it directly in the mutate statement when you declare that column?